src/Bicep.Core/Intermediate/ExpressionVisitor.cs

```diff
@@ -203,10 +207,94 @@ public void VisitUserDefinedFunctionCallExpression(UserDefinedFunctionCallExpres
         Visit(expression.Parameters);
     }
 
+    public void VisitDeclaredTypeExpression(DeclaredTypeExpression expression)
+    {
+        Visit(expression.Value);
```

Comment:
Don't we need to also visit the expressions created for decorators for some of these visit methods?